### PR TITLE
add create/delete op

### PR DESCRIPTION
**Pull request by jkralik**
use device simulator from iotivity-lite repository and add create delete operation

### CODE REVIEW

File: client/getResource_test.go [hunk 1]
```diff
@@ -59,18 +58,24 @@ func TestClientGetResource(t *testing.T) {
...
+	defer func() {
+		err := c.DisownDevice(ctx, deviceID)
+		require.NoError(t, err)
+	}()
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			ctx, cancel := context.WithTimeout(ctx, time.Second)
-			defer cancel()
 			var got map[string]interface{}
-			err := c.GetResource(ctx, tt.args.deviceID, tt.args.href, &got, tt.args.opts...)
```
Comment (Danielius1922) [inline client/getResource_test.go:73]: I think deviceID can be removed from args, since it is no longer used.

File: client/maitenance_test.go [hunk 2]
```diff
@@ -35,17 +34,19 @@ func TestClientFactoryReset(t *testing.T) {
...
+	ctx, cancel := context.WithTimeout(context.Background(), TestTimeout)
+	defer cancel()
+	deviceID, err = c.OwnDevice(ctx, deviceID)
+	require.NoError(t, err)
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			ctx, cancel := context.WithTimeout(context.Background(), time.Second*1)
-			defer cancel()
-			err := c.FactoryReset(ctx, tt.args.deviceID)
+			err := c.FactoryReset(ctx, deviceID)
```
Comment (Danielius1922) [inline client/maitenance_test.go:49]: tt.args.deviceID? Rr the second test case must be modified since it uses different deviceID value

**Review by Danielius1922 — APPROVED** [review]

File: client/getResource_test.go (hunk 1 above)
Comment (jkralik) [reply client/getResource_test.go:73]: I want to test it with non exist device so will leave it and use proper `tt.args.deviceID`

File: client/maitenance_test.go (hunk 2 above)
Comment (jkralik) [reply client/maitenance_test.go]: I want to test it with non exist device so will leave it and use proper `tt.args.deviceID`

File: client/core/client_test.go [hunk 3]
```diff
@@ -22,7 +22,7 @@ import (
 	"github.com/plgd-dev/kit/v2/security"
 )
 
-var TestDeviceName = "devsim-" + test.MustGetHostname()
+//var DevsimName = test.DevsimName
```
Comment (Danielius1922) [inline client/core/client_test.go:25]: Delete?

Comment (jkralik) [reply client/core/client_test.go:25]: deleted

**Pull request merged**